mlflow/server/js/src/experiment-tracking/pages/prompts/hooks/useCreatePromptModal.tsx

```diff
@@ -42,144 +61,221 @@ export const useCreatePromptModal = ({
-        rules={{
-          required: {
-            value: true,
-            message: intl.formatMessage({
-              defaultMessage: 'Prompt content is required',
-              description: 'A validation state for the prompt content in the prompt creation modal',
-            }),
-          },
-        }}
-        placeholder={intl.formatMessage({
-          defaultMessage: "Type prompt content here. Wrap variables with double curly brace e.g. '{{' name '}}'.",
-          description: 'A placeholder for the prompt content in the prompt creation modal',
-        })}
-        validationState={form.formState.errors.draftValue ? 'error' : undefined}
-      />
-      {form.formState.errors.draftValue && (
-        <FormUI.Message type="error" message={form.formState.errors.draftValue.message} />
-      )}
-      <Spacer />
-      <FormUI.Label htmlFor="mlflow.prompts.create.commit_message">Commit message (optional):</FormUI.Label>
-      <RHFControlledComponents.Input
-        control={form.control}
-        id="mlflow.prompts.create.commit_message"
-        componentId="mlflow.prompts.create.commit_message"
-        name="commitMessage"
-      />
-    </Modal>
+        )}
+        {form.watch('promptType') === PROMPT_TYPE_TEXT && form.formState.errors.draftValue && (
+          <FormUI.Message type="error" message={form.formState.errors.draftValue.message} />
+        )}
+        <Spacer />
+        <FormUI.Label htmlFor="mlflow.prompts.create.commit_message">Commit message (optional):</FormUI.Label>
+        <RHFControlledComponents.Input
+          control={form.control}
+          id="mlflow.prompts.create.commit_message"
+          componentId="mlflow.prompts.create.commit_message"
+          name="commitMessage"
+        />
+      </Modal>
+    </FormProvider>
   );
 
   const openModal = () => {
     errorsReset();
     if (mode === CreatePromptModalMode.CreatePromptVersion && latestVersion) {
+      const latestPromptType =
+        latestVersion.tags?.find((tag) => tag.key === PROMPT_TYPE_TAG_KEY)?.value || PROMPT_TYPE_TEXT;
       form.reset({
         commitMessage: '',
         draftName: '',
         draftValue: getPromptContentTagValue(latestVersion) ?? '',
+        chatMessages:
+          latestPromptType === PROMPT_TYPE_CHAT
+            ? (() => {
+                try {
+                  const parsed = JSON.parse(getPromptContentTagValue(latestVersion) ?? '');
+                  return Array.isArray(parsed) ? parsed : [{ role: 'user', content: '' }];
+                } catch {
+                  return [{ role: 'user', content: '' }];
+                }
+              })()
+            : [{ role: 'user', content: '' }],
+        tags: [],
+        promptType: latestPromptType,
+      });
+    } else {
+      form.reset({
+        commitMessage: '',
+        draftName: '',
+        draftValue: '',
+        chatMessages: [{ role: 'user', content: '' }],
         tags: [],
+        promptType: PROMPT_TYPE_TEXT,
```

Comment:
this can be simplified using the proposed typeguard, we probably don't need the `if / else`

```suggestion
      const tagValue = getPromptContentTagValue(latestVersion);
      const parsedValue = parseJSONSafe(tagValue);
      const isChatMessages = isPromptChatMessages(parsedValue);
  
      form.reset({
        commitMessage: '',
        draftName: '',
        draftValue: '',
        chatMessages: isChatMessages ? parsedValue : [{ role: 'user', content: '' }],
        tags: [],
        promptType: isChatMessages ? 'chat' : 'text',
```